src/common/config.h

```diff
@@ -12,181 +12,74 @@
-  }
-
-  inline void ParseStr(std::string *tok) {
-    while ((ch_buf_ = this->GetChar()) != EOF) {
-      switch (ch_buf_) {
-        case '\\': *tok += this->GetChar(); break;
-        case '\"': return;
-        case '\r':
-        case '\n': LOG(FATAL)<< "ConfigReader: unterminated string";
-        default: *tok += static_cast<char>(ch_buf_);
-      }
-    }
-    LOG(FATAL) << "ConfigReader: unterminated string";
-  }
-  inline void ParseStrML(std::string *tok) {
-    while ((ch_buf_ = this->GetChar()) != EOF) {
-      switch (ch_buf_) {
-        case '\\': *tok += this->GetChar(); break;
-        case '\'': return;
-        default: *tok += static_cast<char>(ch_buf_);
-      }
-    }
-    LOG(FATAL) << "unterminated string";
-  }
-  // return newline
-  inline bool GetNextToken(std::string *tok) {
-    tok->clear();
-    bool new_line = false;
-    while (ch_buf_ != EOF) {
-      switch (ch_buf_) {
-        case '#' : SkipLine(); new_line = true; break;
-        case '\"':
-          if (tok->length() == 0) {
-            ParseStr(tok); ch_buf_ = this->GetChar(); return new_line;
-          } else {
-            LOG(FATAL) << "ConfigReader: token followed directly by string";
-          }
-        case '\'':
-          if (tok->length() == 0) {
-            ParseStrML(tok); ch_buf_ = this->GetChar(); return new_line;
-          } else {
-            LOG(FATAL) << "ConfigReader: token followed directly by string";
-          }
-        case '=':
-          if (tok->length() == 0) {
-            ch_buf_ = this->GetChar();
-            *tok = '=';
-          }
-          return new_line;
-        case '\r':
-        case '\n':
-          if (tok->length() == 0) new_line = true;
-        case '\t':
-        case ' ' :
-          ch_buf_ = this->GetChar();
-          if (tok->length() != 0) return new_line;
-          break;
-        default:
-          *tok += static_cast<char>(ch_buf_);
-          ch_buf_ = this->GetChar();
-          break;
-      }
-    }
-    if (tok->length() == 0) {
-      return true;
-    } else {
-      return false;
+      if (line.empty() || name.empty() || value.empty())
+        continue;  // skip a line if # at beginning or there is no value or no name.
+      CleanString(&name);  // clean the string
+      CleanString(&value);
+      results.emplace_back(name, value);
     }
+    return results;
   }
-};
-/*!
- * \brief an iterator use stream base, allows use all types of istream
- */
-class ConfigStreamReader: public ConfigReaderBase {
- public:
-  /*!
-   * \brief constructor
-   * \param fin istream input stream
-   */
-  explicit ConfigStreamReader(std::istream &fin) : fin_(fin) {}
 
- protected:
-  int GetChar() override {
-    return fin_.get();
-  }
-  /*! \brief to be implemented by child, check if end of stream */
-  bool IsEnd() override {
-    return fin_.eof();
+  ~ConfigParse() {
+    fi_.close();
   }
 
  private:
-  std::istream &fin_;
-};
+  std::ifstream fi_;
+  std::string allowableChar =
```

Comment:
I'm curious why clang-tidy didn't catch this styling error.

Reply:
@trivialfis I don't see any issue with this line?

Reply:
@hcho3 private member suffix '_'.  I tried clang-tidy-8 and it didn't catch it either.